# Assigment - Part 2, Making and populating a Database

In [1]:
import pandas as pd
df = pd.read_csv('https://raw.githubusercontent.com/gyhou/DS-Unit-3-Sprint-2-SQL-and-Databases/master/module1-introduction-to-sql/buddymove_holidayiq.csv')
print(df.shape)
df.head()

(249, 7)


User Id  Sports  Religious  Nature  Theatre  Shopping  Picnic
0  User 1       2         77      79       69        68      95
1  User 2       2         62      76       76        69      68
2  User 3       2         50      97       87        50      75
3  User 4       2         68      77       95        76      61
4  User 5       2         98      54       59        95      86

In [31]:
df.describe()

Sports   Religious      Nature     Theatre    Shopping      Picnic
count  249.000000  249.000000  249.000000  249.000000  249.000000  249.000000
mean    11.987952  109.779116  124.518072  116.377510  112.638554  120.401606
std      6.616501   32.454115   45.639372   32.132696   41.562888   32.633339
min      2.000000   50.000000   52.000000   59.000000   50.000000   61.000000
25%      6.000000   84.000000   89.000000   93.000000   79.000000   92.000000
50%     12.000000  104.000000  119.000000  113.000000  104.000000  119.000000
75%     18.000000  132.000000  153.000000  138.000000  138.000000  143.000000
max     25.000000  203.000000  318.000000  213.000000  233.000000  218.000000

In [2]:
import sqlite3
conn = sqlite3.connect('buddymove_holidayiq.sqlite3')
df.to_sql('review', con=conn)
curs = conn.cursor()

/usr/local/lib/python3.6/dist-packages/pandas/core/generic.py:2531: UserWarning: The spaces in these column names will not be changed. In pandas versions < 0.14, spaces were converted to underscores.
  dtype=dtype, method=method)


In [4]:
query = """SELECT * FROM review LIMIT 5"""
curs.execute(query).fetchall()

[(0, 'User 1', 2, 77, 79, 69, 68, 95),
 (1, 'User 2', 2, 62, 76, 76, 69, 68),
 (2, 'User 3', 2, 50, 97, 87, 50, 75),
 (3, 'User 4', 2, 68, 77, 95, 76, 61),
 (4, 'User 5', 2, 98, 54, 59, 95, 86)]

### Count how many rows you have - it should be 249!

In [5]:
query = """
SELECT count(0)
FROM review
"""
curs.execute(query).fetchall()

[(249,)]

### How many users who reviewed at least 100 Nature in the category also reviewed at least 100 in the Shopping category?

- 78 users

In [6]:
query = """
SELECT count(Sports)
FROM review
WHERE Nature >= 100
AND Shopping >= 100
"""
curs.execute(query).fetchall()

[(78,)]

### (Stretch) What are the average number of reviews for each category?

| Sports | Religious | Nature | Theatre | Shopping | Picnic |
|--------|-----------|--------|---------|----------|--------|
| 11.99  | 109.78    | 124.52 | 116.38  | 112.64   | 120.40 |

In [7]:
query = """
SELECT avg(Sports),
avg(Religious),
avg(Nature),
avg(Theatre),
avg(Shopping),
avg(Picnic)
FROM review
"""
curs.execute(query).fetchall()

[(11.987951807228916,
  109.77911646586345,
  124.51807228915662,
  116.37751004016064,
  112.63855421686748,
  120.40160642570281)]